### ISSUE TITLE
Conflicting marker warnings for *** and ___ strong emphasis

### ISSUE THREAD

**Issue by spl** [remarkjs/remark-lint#236]
### Subject of the issue

* CommonMark [Example 466](https://spec.commonmark.org/0.29/#example-466) demonstrates the use of `***foo***` to produce `<em><strong>foo</strong></em>`.
* Using `remark-lint-strong-marker` and `remark-lint-emphasis-marker` together with different options (`'*'` and `'_'` in my case) results in warnings for both `***foo***` and `___foo___`.
* Each warning cannot be resolved by changing `***` to `___` or vice versa.
* It seems like `**_foo_**` and `_**foo**_` serve as a workaround for this problem.

### Your environment

*   **OS**: macOS 10.14.6 (Mojave)
*   **Packages**: See `package.json` below.
*   **Env**: npm 6.14.4, node 14.3.0

### Steps to reproduce

Run the `index.js` below.

```sh
$ node index.js 
```

### Expected behaviour

I didn't expect a warning. I also didn't expect to be unable to write either `***foo***` or `___foo___`.

### Actual behaviour

```sh
  1:3-1:8  warning  Emphasis should use `_` as a marker  emphasis-marker  remark-lint

⚠ 1 warning
  1:1-1:10  warning  Strong should use `*` as a marker  strong-marker  remark-lint

⚠ 1 warning
no issues found
no issues found
```

### Attachments

`index.js`:

```javascript
const remark = require('remark')()
  .use(require('remark-lint-strong-marker'), '*')
  .use(require('remark-lint-emphasis-marker'), '_')
const report = require('vfile-reporter')

function process(s) {
  console.log(report(remark.processSync(s)))
}

process('***foo***')
process('___foo___')
process('**_foo_**')
process('_**foo**_')
```

`package.json`:

```json
{
  "name": "tmp",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "remark": "12.0.0",
    "remark-lint": "7.0.0",
    "remark-lint-emphasis-marker": "2.0.0",
    "remark-lint-strong-marker": "2.0.0",
    "vfile-reporter": "6.0.1"
  }
}
```

**Comment by wooorm** [comment]
If you're preferring different markers for the different types, and you get messages if one is used, then, well, it seems to be working? Why do you expect three asterisks or three underscores to not show a lint warning?

Note: CM shows many weird edge cases, so there are many more things that work in markdown but this project warns about!

**Comment by spl** [comment]
I would conjecture that (a) this is an edge case in Markdown itself, (b) CommonMark provides a standardized solution, and (c) the use of the two different lint packages for markers may suggest that it might be better to use one package for strong and emphasis markers.

That said, I understand if this issue is not considered worth fixing. At least there are alternative ways of stating the same thing.

**Comment by spl** [comment]
> If you're preferring different markers for the different types, and you get messages if one is used, then, well, it seems to be working? Why do you expect three asterisks or three underscores to not show a lint warning?

On reflection, I realize now that perhaps I wasn't thinking clearly earlier. I was considering `***` to be a single markup opener that represented a combination of `<strong>` and `<em>`. This note in the [spec](https://spec.commonmark.org/0.29/#emphasis-and-strong-emphasis) likely contributed to my confusion:

> An interpretation `<em><strong>...</strong></em>` is always preferred to `<strong><em>...</em></strong>`.

Instead, I should've been thinking of it as `*` `**` for `<em>` `<strong>`. In which case, `remark-lint` warns for the symbol that is unexpected. Thus, as you point out, it is working.

Thanks, and sorry for the false report!

**Comment by wooorm** [comment]
Ah, yes that indeed was a given for me, and now it makes it more clear to me what you wrote. Indeed: they’re separate! You can also go deeper than emphasis, strong, strong emphasis!
